File: src/main/kotlin/samsung/test03.java
package samsung;

// 시험 감독

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class test03 {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        int[] a = new int[n];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(st.nextToken());
        }

        st = new StringTokenizer(br.readLine());
        int b = Integer.parseInt(st.nextToken());
        int c = Integer.parseInt(st.nextToken());

        long answer = 0;
        for (int i = 0; i < n; i++) {
            a[i] -= b;
            answer++;

            int num = (int) Math.ceil((double) a[i] / c);
            if (num > 0) {
                answer += num;
            }
        }


        System.out.println(answer);
    }

}

File: src/main/kotlin/nadongbin/greedy/test03.kt
package nadongbin.greedy

// 1이 될 때까지

import java.io.BufferedReader
import java.io.InputStreamReader

private fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    var (n, k) = readLine().split(" ").map { it.toInt() }

    var answer = 0

    while(n > 1){
        if(n % k == 0){
            n /= k
        }else{
            n --
        }
        answer++
    }

    println(answer)
}
